controllers/usersignup/usersignup_controller.go

```diff
@@ -377,8 +403,21 @@ func (r *Reconciler) ensureNewMurIfApproved(reqLogger logr.Logger, config toolch
 		return r.wrapErrorWithStatusUpdate(reqLogger, userSignup, r.setStatusNoTemplateTierAvailable, err, "")
 	}
 
+	// Set the last-target-cluster annotation so that if the user signs up again later on, they can be provisioned to the same cluster
+	userSignup.Annotations[toolchainv1alpha1.UserSignupLastTargetClusterAnnotationKey] = targetCluster.getClusterName()
+	if err := r.Client.Update(context.TODO(), userSignup); err != nil {
+		return r.wrapErrorWithStatusUpdate(reqLogger, userSignup, r.setStatusFailedToUpdateAnnotation, err,
+			"unable to update last target cluster annotation on UserSignup resource")
+	}
+
+	compliantUsername, err := r.generateCompliantUsername(config, userSignup)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(reqLogger, userSignup, r.setStatusFailedToCreateMUR, err,
+			"Error generating compliant username for %s", userSignup.Spec.Username)
+	}
```

Comment:
it's just a detail, but this could have been moved back to `provisionMasterUserRecord` method.

Reply:
Right, good catch! Done in d352f8535af73d6d525e8323aca3159d6c54f6b6